File: src/pers/gargantua/mine_sweep/MainFrame.java
package pers.gargantua.mine_sweep;

import java.awt.BorderLayout;
import java.awt.CardLayout;

import javax.swing.JFrame;
import javax.swing.JPanel;

public final class MainFrame extends JFrame{
	
	public static void main(String[] args) {
		MainFrame.getMainFrame();
	}
	
	private static MainFrame mainFrame = null;
	
	public final JPanel mainPanel;
	public final CardLayout cardLayout = new CardLayout();
	
	private MainFrame() {
		
		JPanel gamePanel = new JPanel();
		gamePanel.setLayout(new BorderLayout());
		gamePanel.add(Gaming.getGaming(), BorderLayout.CENTER);
		gamePanel.add(StaBar.getStaBar(), BorderLayout.SOUTH);
		
		mainPanel = new JPanel();
		mainPanel.setLayout(cardLayout);
		mainPanel.add("start", Start.getStart());
		mainPanel.add("gaming", gamePanel);
		
		setLayout(new BorderLayout());
		
		add(new Menu(), BorderLayout.NORTH);
		add(mainPanel, BorderLayout.CENTER);
		
		setTitle("Mine Sweep");
		setSize(300, 230);
		setResizable(false);
		setLocationRelativeTo(null);
		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		setVisible(true);
	}
	
	public static MainFrame getMainFrame() {
		if(mainFrame == null)
			mainFrame = new MainFrame();
		return mainFrame;
	}
	
}

File: src/pers/gargantua/mine_sweep/MainFrame.kt
package pers.gargantua.mine_sweep

import java.awt.BorderLayout
import java.awt.CardLayout
import javax.swing.JFrame
import javax.swing.JPanel


/**
 * @author Gargantua丶
 **/

object MainFrame : JFrame() {

    @JvmStatic
    fun main(args: Array<String>) {
        MainFrame
    }

    val mainPanel: JPanel = JPanel()
    private val gamePanel: JPanel = JPanel()
    val cardLayout = CardLayout()

    init {
        gamePanel.apply {
            layout = BorderLayout()
            add(Gaming, BorderLayout.CENTER)
            add(StaBar, BorderLayout.SOUTH)
        }

        mainPanel.apply {
            layout = cardLayout
            add("start", Start)
            add("gaming", gamePanel)
        }

        layout = BorderLayout()

        add(Menu, BorderLayout.NORTH)
        add(mainPanel, BorderLayout.CENTER)

        title = "Mine Sweep"
        setSize(300, 230)
        isResizable = false
        setLocationRelativeTo(null)
        defaultCloseOperation = EXIT_ON_CLOSE
        isVisible = true
        cardLayout.show(mainPanel, "start")
    }
}